### PR TITLE
Remove `JSXText` spacial case in comment attaching

### PR DESCRIPTION
**Pull request by fisker**
## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/common/util.js [hunk 1]
```diff
@@ -565,14 +565,6 @@ function addCommentHelper(node, comment) {
   const comments = node.comments || (node.comments = []);
   comments.push(comment);
   comment.printed = false;
-
-  // For some reason, TypeScript parses `// x` inside of JSXText as a comment
```
Comment (thorn0) [inline src/common/util.js:569]: Does this still happen though? I can't reproduce.

Comment (thorn0) [reply src/common/util.js:569]: I guess this has been fixed by https://github.com/typescript-eslint/typescript-eslint/pull/703

Comment (fisker) [reply src/common/util.js:569]: I think this comment mean

```js
<Foo>
  text
  // comment
  text
</Foo>
```

`// comment` is parsed as comment? I checked other parsers they parse it as 

```js
{
              "type": "JSXText",
              "start": 5,
              "end": 33,
              "loc": {
                "start": {
                  "line": 1,
                  "column": 5
                },
                "end": {
                  "line": 5,
                  "column": 0
                }
              },
              "extra": {
                "rawValue": "\n  text\n  // comment\n  text\n",
                "raw": "\n  text\n  // comment\n  text\n"
              },
              "value": "\n  text\n  // comment\n  text\n"
            }
```

And TypeScript parses it as

```
          {
            "type": "Literal",
            "value": "\n  text\n  // comment\n  text\n",
            "raw": "\n  text\n  // comment\n  text\n",
            "range": [
              5,
              33
            ]
          }
```

Anyway, I think you are right, it's not need anymore

Comment (thorn0) [reply src/common/util.js:569]: That's a separate thing controlled by the `useJSXTextNode` option of `typescript-estree`. Prettier uses `useJSXTextNode: true`.

**Review by thorn0 — APPROVED** [review]

**Pull request merged**